### ISSUE TITLE
Evolve fails if current schema is not set

### ISSUE THREAD

**Issue by tehmantra** [lecaillon/Evolve#156]
Hi,

First of all, thanks for creating Evolve, it has been able to solve SQL deployments, which have been a huge pain point for our company for many years!

### Platform:
* evolve.tool 2.3.0
* Microsoft SQL Server 2016

### Description:
In SQL Server, when using Windows Groups as a database user, it is possible to set the `DEFAULT_SCHEMA` to `NULL`
e.g.
```sql
ALTER USER [MYDOMAIN\SomeGroup] WITH DEFAULT_SCHEMA = NULL;
```

When a member of this group logs into the database, `SELECT SCHEMA_NAME()` will return `NULL`.

If that user tries to run evolve migrate, it will fail:

```
$ evolve migrate sqlserver --schema MySchema --connection-string "Server=...; Database=...; Integrated Security=true;"

Executing Migrate...
Unable to cast object of type 'System.DBNull' to type 'System.String'. Sql query: SELECT SCHEMA_NAME() Unable to cast object of type 'System.DBNull' to type 'System.String'.
Unable to cast object of type 'System.DBNull' to type 'System.String'. Sql query: SELECT SCHEMA_NAME() Unable to cast object of type 'System.DBNull' to type 'System.String'.
```

This happens because in SQLServerDatabase.cs:105 we run the following
```csharp
public override string GetCurrentSchemaName() => WrappedConnection.QueryForString("SELECT SCHEMA_NAME()");
```

Which calls QueryForString:
```csharp
public static string QueryForString(this WrappedConnection wrappedConnection, string sql)
{
    return Execute(wrappedConnection, sql, cmd =>
    {
        return (string)cmd.ExecuteScalar();
        //     ^^^^^^^^ cast fails
    });
}
```
_________

It seems like we could naively fix this by checking for `DBNull.Value` and returning a null string, but that will probably cause failures elsewhere. e.g. `defaultSchema.Equals(...`.

**Comment by lecaillon** [comment]
Hi @tehmantra  and thanks for your feedback. Always a pleasure to know Evolve is usefull somewhere for someone :)

**Comment by lecaillon** [comment]
Fixed. 
Evolve still needs to find one shema in order to create/locate the metadatatable via the `Evolve.Schemas ` option, the `Evolve.MetadataTableSchema` option or the datasource connection. Otherwise it will raises a proper `EvolveConfigurationException`.

**Comment by tehmantra** [comment]
Just confirmed this is fixed by 2.4.0. Many thanks.
